lib/chef/provider/aws_ebs_volume.rb

```diff
@@ -29,12 +39,67 @@ class Chef::Provider::AwsEbsVolume < Chef::Provider::AwsProvider
     if existing_volume
       converge_by "Deleting EBS volume #{fqn} in #{new_resource.region_name}" do
         existing_volume.delete
+
+        Retryable.retryable(:tries => 30, :sleep => 2, :on => TimeoutError) do
+          raise TimeoutError,
+            "Timed out waiting for volume #{new_resource.volume_id} to be deleted." if existing_volume.exists?
+        end
       end
     end
 
     new_resource.delete
   end
 
+  action :attach do
```

Comment:
Actually, if you get a chance you might want to look at the jk/no_mo_data_bag_resource branch.  I may have already made machine work there, I did a general sweep through all the resources.